src/RpcClient.ts

```diff
@@ -646,81 +650,118 @@ export class RpcClient {
     // Rebuild the transaction with the final instructions
     if (isVersioned) {
       const v0Message = new TransactionMessage({
-        instructions: instructions,
-        payerKey: payerKey,
-        recentBlockhash: recentBlockhash,
+        instructions,
+        payerKey,
+        recentBlockhash,
       }).compileToV0Message(lookupTables);
 
       versionedTransaction = new VersionedTransaction(v0Message);
 
       const allSigners = feePayer ? [...signers, feePayer] : signers;
       versionedTransaction.sign(allSigners);
 
-      return { smartTransaction: versionedTransaction, lastValidBlockHeight };
-    } else {
-      legacyTransaction = new Transaction().add(...instructions);
-      legacyTransaction.recentBlockhash = recentBlockhash;
-      legacyTransaction.feePayer = payerKey;
-
-      for (const signer of signers) {
-        legacyTransaction.partialSign(signer);
-      }
+      return {
+        txBuff: versionedTransaction.serialize(),
```

Comment:
I'd prefer if we made the serialization explicit before we return. Also, the name `txBuffer` implies it's a buffer, which is what serialization produces, so we should have it serialized beforehand. It's also arguably safer to do so